src/pykeen/models/unimodal/trans_h.py

```diff
@@ -90,47 +90,39 @@ def __init__(
         self.scoring_fct_norm = scoring_fct_norm
 
         # embeddings
-        self.normal_vector_embeddings = get_embedding(
+        self.normal_vector_embeddings = Embedding.init_with_device(
             num_embeddings=triples_factory.num_relations,
             embedding_dim=embedding_dim,
             device=self.device,
+            # Normalise the normal vectors by their l2 norms
+            constrainer=functional.normalize,
         )
 
-    def _reset_parameters_(self):  # noqa: D102
-        for emb in [
-            self.entity_embeddings,
-            self.relation_embeddings,
-            self.normal_vector_embeddings,
-        ]:
-            emb.reset_parameters()
-        # TODO: Add initialization
-
     def post_parameter_update(self) -> None:  # noqa: D102
-        # Make sure to call super first
         super().post_parameter_update()
+        self.normal_vector_embeddings.post_parameter_update()
 
-        # Normalise the normal vectors by their l2 norms
-        functional.normalize(
-            self.normal_vector_embeddings.weight.data,
-            out=self.normal_vector_embeddings.weight.data,
-        )
+    def _reset_parameters_(self):  # noqa: D102
+        super()._reset_parameters_()
+        self.normal_vector_embeddings.reset_parameters()
+        # TODO: Add initialization
```

Comment:
can we clarify what initialization is necessary here?

Reply:
This TODO was already existent before, so I am not 100% sure why it is there in the first place.
I suppose it is about explicitly initialising the `normal_vector_embeddings`, in case the pytorch default would change in the future.